app/components/Form/utils.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import { SubmissionError } from 'redux-form';
+import { get } from 'lodash';
+
+/*
+ * Simple utility that handles submission errors
+ *
+ * Usage:
+ * onSubmit(data).catch(handleSubmissionError)
+ * or
+ * onSubmit(data).then(result=>{...}, handleSubmissionError)
+ */
+export const handleSubmissionError = (error: any) => {
+  const errPayload = get(error, ['payload', 'response', 'jsonData']);
+  if (!errPayload) {
+    throw error;
+  }
+
+  throw new SubmissionError({
+    ...errPayload,
+    _error: errPayload.detail
```

Comment:
I'm not sure I get this. Isn't this payload included by default when `error` is thrown by itself?

Reply:
Added a comment about this in the `Props` section.